test/test_train_cifar.py

```diff
@@ -122,26 +122,31 @@ def train_cifar():
                              (0.2023, 0.1994, 0.2010)),
     ])
 
-    trainset = torchvision.datasets.CIFAR10(
+    train_dataset = torchvision.datasets.CIFAR10(
         root=FLAGS.datadir,
         train=True,
         download=True,
         transform=transform_train)
-    train_loader = torch.utils.data.DataLoader(
-        trainset,
-        batch_size=FLAGS.batch_size,
-        shuffle=True,
-        num_workers=FLAGS.num_workers)
-
-    testset = torchvision.datasets.CIFAR10(
+    test_dataset = torchvision.datasets.CIFAR10(
         root=FLAGS.datadir,
         train=False,
         download=True,
         transform=transform_test)
+    train_sampler = torch.utils.data.distributed.DistributedSampler(
```

Comment:
I'd feel better if this is done if world_size > 1.
IOW, pass `sampler=None` (or whatever default) if world_size==1.

Reply:
Done. Tested on all 3 datasets and v3-8/v3-32.